### ISSUE TITLE
Debug mode builds of Flang should properly handle -fsyntax-only option

### ISSUE THREAD

**Issue by pawosm-arm** [flang-compiler/clang#8]
The -fsyntax-only option is very popular for testing clang. When applied to debug mode build of Flang it causes assertion fail:
```
tools/clang/lib/Frontend/FrontendAction.cpp:265: bool clang::FrontendAction::BeginSourceFile(clang::CompilerInstance&, const clang::FrontendInputFile&): Assertion `hasIRSupport() && "This action does not have IR file support!"' failed.
```
Does not occur with Release mode builds of Flang.

**Comment by sscalpone** [comment]
From https://github.com/ThePortlandGroup/clang/issues/3:

The -fsyntax-only option is very popular for testing clang. When applied to flang it causes assertion fail:
tools/clang/lib/Frontend/FrontendAction.cpp:261: bool clang::FrontendAction::BeginSourceFile(clang::CompilerInstance&, const clang::FrontendInputFile&): Assertion `hasIRSupport() && "This action does not have IR file support!"' failed.

This tend to happened only with LLVM and Clang built in Debug mode.

**Comment by pawosm-arm** [comment]
Can you update release_50 branch too? We're working only on this branch nowadays.

**Comment by gklimowicz** [comment]
Closed with #36.
